server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -255,7 +255,7 @@ public PublicationContext newPublicationContext(ClusterChangedEvent clusterChang
 
     private static BytesReference serializeFullClusterState(ClusterState clusterState, Version nodeVersion) throws IOException {
         final BytesStreamOutput bStream = new BytesStreamOutput();
-        try (StreamOutput stream = new OutputStreamStreamOutput(CompressorFactory.COMPRESSOR.threadLocalOutputStream(bStream))) {
+        try (StreamOutput stream = new OutputStreamStreamOutput(CompressorFactory.DEFLATE_COMPRESSOR.threadLocalOutputStream(bStream))) {
```

Comment:
It is a great idea, but the problem is that in case of mapping and binary fields OS uses only deflate compressor and does not set any magic byte(s) to determine what compression type was used. Maybe it would be better to crate an issue about it and prepare an improvement?

Reply:
But won't it set the header ? Sorry, which magic bytes you mean?

Reply:
fixed